java/core/src/main/java/sleeper/core/record/process/status/ProcessRun.java

```diff
@@ -19,17 +19,17 @@
 import sleeper.core.record.process.RecordsProcessedSummary;
 
 import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
 import java.util.Objects;
 
 public class ProcessRun {
     private final String taskId;
-    private final ProcessRunStartedUpdate startedStatus;
-    private final ProcessFinishedStatus finishedStatus;
```

Comment:
It might still be worth holding the started and finished statuses in fields, rather than scanning through the list every time. I don't feel strongly about that though.